Framework/Crystal/inc/MantidCrystal/FindSXPeaks.h

```diff
@@ -69,17 +84,6 @@ struct DLLExport SXPeak {
   @return q vector
   */
   Mantid::Kernel::V3D getQ() const {
-    auto samplePos = _inst->getSample()->getPos();
-    auto sourcePos = _inst->getSource()->getPos();
-    auto detPos = _inst->getDetector(_detectorId)->getPos();
```

Comment:
Looking at this gain, I think this class should actually use `DetectorInfo`, not `SpectrumInfo`, if we want to keep it consistent with the old behavior (but read also below first!). The index you pass would then be an index of the detector (obtained with `DetectorInfo::indexOf(detid_t)`).